server/apply.go

```diff
@@ -0,0 +1,132 @@
+package server
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/skuid/helm-value-store/store"
+	"go.uber.org/zap"
+)
+
+type applyRequest struct {
+	UUID string `json:"uuid"`
+}
+
+type applyResponse struct {
+	Status  string `json:"status"`
+	Message string `json:"message"`
+}
+
+type upsertConfig struct {
+	location string
+	timeout  int64
+}
+
+func upsertRelease(r *store.Release, conf upsertConfig) error {
+	_, err := r.Get()
+
+	if err != nil && !strings.Contains(err.Error(), "not found") {
+		return err
+	}
+
+	if err != nil && strings.Contains(err.Error(), "not found") {
+		_, err = r.Install(conf.location, false, conf.timeout)
+	} else if err == nil {
+		_, err = r.Upgrade(conf.location, false, conf.timeout)
+	}
+
+	return err
+}
+
+func (c ApiController) ApplyChart(w http.ResponseWriter, r *http.Request) {
+	var err error
+	if r.Method != http.MethodPost {
+		w.WriteHeader(http.StatusNotFound)
+		return
+	}
+
+	// Fields for the autit log
+	auditFields := c.authorizer.LoggingClosure(r)
+	defer func() {
+		var successful bool
+		if err == nil {
+			successful = true
+		}
+		auditFields = append(
+			auditFields,
+			zap.String("controller", "apply"),
+			zap.Bool("successful", successful),
+		)
+		zap.L().Info("Audit Log", auditFields...)
+	}()
+
+	applyReq := &applyRequest{}
+
+	if err = json.NewDecoder(r.Body).Decode(applyReq); err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		zap.L().Error("Error decoding request", zap.Error(err))
+		return
+	}
+	auditFields = append(auditFields, zap.String("uuid", applyReq.UUID))
+
+	release := &store.Release{}
+
+	release, err = c.releaseStore.Get(r.Context(), applyReq.UUID)
+
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
```

Comment:
i feel like we should write _something_ back here so that you don't have to go to the logs whenever you get a 400.